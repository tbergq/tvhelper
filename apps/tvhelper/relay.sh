#!/bin/bash

cp ../graphql/schema.graphql ./schema.graphql
yarn relay-compiler --language typescript