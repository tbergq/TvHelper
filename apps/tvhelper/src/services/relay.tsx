import { ReactNode } from 'react';
import { Environment, Network, RecordSource, Store, FetchFunction } from 'relay-runtime';
import { RelayEnvironmentProvider } from 'react-relay';
import { RecordMap } from 'relay-runtime/lib/store/RelayStoreTypes';

interface Props {
  readonly children: ReactNode;
  readonly records?: RecordMap;
  readonly token: string | undefined;
}

interface CreateEnvironmentArgs {
  fetchQuery: FetchFunction;
  records?: RecordMap;
}
export const createEnvironment = ({ fetchQuery, records }: CreateEnvironmentArgs) => {
  const network = Network.create(fetchQuery);

  const store = new Store(new RecordSource(records));

  return new Environment({
    network,
    store,
  });
};

const { GRAPHQL_URL } = process.env;

const getBody = (operation, variables) => {
  if (operation.id) {
    return {
      queryId: operation.id,
      variables,
    };
  }
  return {
    query: operation.text,
    variables,
  };
};

export const makeFetchQuery = (token, baseUrl = '') => {
  const fetchQuery = async (operation, variables /* , cacheConfig, uploadables */) => {
    const res = await fetch(`${baseUrl}`, {
      method: 'POST',
      headers: {
        // Add authentication and other headers here
        'content-type': 'application/json',
        'authorization': token,
      },
      body: JSON.stringify(getBody(operation, variables)),
    });
    return res.json();
  };
  return fetchQuery;
};

const EnvironmentProvider = ({ children, records, token }: Props) => {
  const environment = createEnvironment({
    fetchQuery: makeFetchQuery(token, GRAPHQL_URL),
    records,
  });

  return <RelayEnvironmentProvider environment={environment}>{children}</RelayEnvironmentProvider>;
};

export { EnvironmentProvider };
