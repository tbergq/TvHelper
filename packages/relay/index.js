// @flow

export { default as QueryRenderer } from './src/QueryRenderer';
export { graphql, createFragmentContainer } from '@kiwicom/relay';
