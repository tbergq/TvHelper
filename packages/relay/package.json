{
  "name": "@tbergq/relay",
  "private": true,
  "main": "index.js",
  "version": "0.0.1",
  "sideEffects": false,
  "dependencies": {
    "@adeira/fetch": "^1.0.2",
    "@adeira/js": "^1.2.1",
    "@adeira/relay": "^1.3.0",
    "@adeira/relay-utils": "^0.7.0",
    "@tbergq/components": "*",
    "graphql": "^15.0.0",
    "react-fast-compare": "^3.1.1"
  },
  "peerDependencies": {
    "react": "",
    "react-relay": "*",
    "relay-runtime": ""
  },
  "devDependencies": {
    "relay-test-utils": "^9.1.0"
  }
}
