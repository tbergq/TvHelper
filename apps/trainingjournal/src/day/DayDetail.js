// @flow

import * as React from 'react';
import { createFragmentContainer, graphql } from '@tbergq/relay';
import { Heading } from '@tbergq/components';

import type { DayDetail_day as DayType } from './__generated__/DayDetail_day.graphql';
import ExerciseTable from './ExerciseTable';

type Props = {|
  +day: ?DayType,
|};

const DayDetail = (props: Props) => {
  const dayName = props.day?.name ?? '';
  return (
    <>
      <Heading>{dayName}</Heading>
      <ExerciseTable day={props.day} />
    </>
  );
};

export default createFragmentContainer(DayDetail, {
  day: graphql`
    fragment DayDetail_day on Day {
      name
      ...ExerciseTable_day
    }
  `,
});
