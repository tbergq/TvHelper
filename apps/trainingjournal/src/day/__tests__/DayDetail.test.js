// @flow

import * as React from 'react';
import { create } from 'react-test-renderer';
import { createMockEnvironment, MockPayloadGenerator } from 'relay-test-utils';
import { QueryRenderer, graphql } from '@kiwicom/relay';

import DayDetail from '../DayDetail';

const renderer = props => <DayDetail day={props.day} />;

it('renders', () => {
  const environment = createMockEnvironment();
  const TestRenderer = () => (
    <QueryRenderer
      environment={environment}
      query={graphql`
        query DayDetailQuery @relay_test_operation {
          day(dayId: "123") {
            ...DayDetail_day
          }
        }
      `}
      onResponse={renderer}
    />
  );
  const wrapper = create(<TestRenderer />);
  environment.mock.resolveMostRecentOperation(operation =>
    MockPayloadGenerator.generate(operation, {
      Exercise: () => ({
        set: '4',
        reps: '5',
        breakTime: '2',
      }),
      BaseExercise: () => ({
        name: 'Kneb√∏y',
      }),
    }),
  );
  expect(wrapper).toMatchSnapshot();
});
