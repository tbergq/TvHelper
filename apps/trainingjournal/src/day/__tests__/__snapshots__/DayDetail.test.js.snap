// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  <h1
    className="Heading__StyledHeading-sc-1b8cso5-0 kCqDso"
  >
    &lt;mock-value-for-field-"name"&gt;
  </h1>,
  <div
    className="Table__StyledTableOuter-sc-156ovmd-0 eJfAYr"
  >
    <div
      className="Table__StyledTableInner-sc-156ovmd-1 jsHlsU"
      onScroll={[Function]}
    >
      <table
        className="Table__StyledTable-sc-156ovmd-2 fCREQU"
      >
        <thead
          className="TableHead__StyledTableHead-sc-17bka47-0 kXbKtX"
        >
          <tr
            className="TableRow__StyledTableRow-nnt9s9-0 efbCJT"
          >
            <td
              className="TableCell__StyledTableCell-sc-1dr7ow2-0 jTONqK"
            >
              Exercise
            </td>
            <td
              className="TableCell__StyledTableCell-sc-1dr7ow2-0 jTONqK"
            >
              Sets
            </td>
            <td
              className="TableCell__StyledTableCell-sc-1dr7ow2-0 jTONqK"
            >
              Reps
            </td>
            <td
              className="TableCell__StyledTableCell-sc-1dr7ow2-0 jTONqK"
            >
              Break
            </td>
          </tr>
        </thead>
        <tbody
          className="TableBody__StyledTableBody-sc-1uovf8k-0 kqMaSS"
        >
          <tr
            className="TableRow__StyledTableRow-nnt9s9-0 efbCJT"
          >
            <td
              className="TableCell__StyledTableCell-sc-1dr7ow2-0 jTONqK"
            >
              Kneb√∏y
            </td>
            <td
              className="TableCell__StyledTableCell-sc-1dr7ow2-0 jTONqK"
            >
              4
            </td>
            <td
              className="TableCell__StyledTableCell-sc-1dr7ow2-0 jTONqK"
            >
              5
            </td>
            <td
              className="TableCell__StyledTableCell-sc-1dr7ow2-0 jTONqK"
            >
              2
               min
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
]
`;
